ocaml/xapi/xapi_pool.ml

```diff
@@ -3659,3 +3659,38 @@ let reset_telemetry_uuid ~__context ~self =
   Xapi_stdext_pervasives.Pervasiveext.ignore_exn (fun _ ->
       Db.Secret.destroy ~__context ~self:old_ref
   )
+
+let configure_update_sync ~__context ~self ~update_sync_frequency
+    ~update_sync_day =
+  ( match (update_sync_frequency, update_sync_day) with
+  | `daily, d when d <> 0L ->
+      error
+        "For daily schedule, cannot set the day when update sync will run to \
+         an integer other than 0" ;
+      raise
+        Api_errors.(
+          Server_error
+            (invalid_update_sync_day, [Int64.to_string update_sync_day])
+        )
+  | `weekly, d when d < 0L || d > 6L ->
+      error
+        "For weekly schedule, cannot set the day when update sync will run to \
+         an integer out of range: 0 ~ 6" ;
+      raise
+        Api_errors.(
+          Server_error
+            (invalid_update_sync_day, [Int64.to_string update_sync_day])
+        )
+  | _ ->
+      ()
+  ) ;
+  Db.Pool.set_update_sync_frequency ~__context ~self
+    ~value:update_sync_frequency ;
+  Db.Pool.set_update_sync_day ~__context ~self ~value:update_sync_day ;
+  if Db.Pool.get_update_sync_enabled ~__context ~self then
+    (* re-schedule periodic update sync with new configuration immediately *)
+    Xapi_pool_helpers.PeriodicUpdateSync.set_enabled ~__context ~value:true
+
+let set_update_sync_enabled ~__context ~self ~value =
+  Db.Pool.set_update_sync_enabled ~__context ~self ~value ;
```

Comment:
Would suggest to set the DB in the last step.
Otherwise, in failure case of `Xapi_pool_helpers.PeriodicUpdateSync.set_enabled`, the DB would have been changed inconsistently.